### PR TITLE
fix: rent formatting hack

### PR DESCRIPTION
**Pull request by ColinBuyck**
# Pull Request Template

## Issue Overview

This PR addresses [#15](https://github.com/bloom-housing/Bloom-Doorway/issues/15)

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

These changes address the fact that rent fields were accepting the negative sign and "e" along with changing via scroll. Also, it was allowing decimal points but that would also break the filtering results.

**Disclaimer:** This is undoubtedly a hacky approach and a more sustainable approach should be implemented [here](https://github.com/bloom-housing/Bloom-Doorway/issues/16). The difficulty came from the fact that our currency formatting utilizes react hook forms but these two forms were made with a custom react state form object and their own updateValue functions. This was the quickest way I found to integrate our formatting with their value setting. Other approaches such as introducing the formatting manually within the onChange function still allows the characters to be typed in the field OR having both react hook forms setValue and their onChange proved that the setValue overrides the onChange. Lastly, I considered handling this solely via react hook forms but that resulted in either too much time required to update the styled button groups to field groups since the typing has to be refactored as well or having two entirely separate form objects which felt more unnatural.

## How Can This Be Tested/Reviewed?

This can be tested by pulling this down locally, go to the landing page, trying to enter negative values or letters unsuccessfully, and noticing how the rent formats when clicking out of the field. Then click view listings and observe that the results and filter reflect the previously entered field. Then go into the listings search modal and perform the same tests.

Note: There's an edge case that if you type in a value like 1200.23 into the landing page, it will show results for 1200 and when opening the listings search modal it will only show 1200. The url search doesn't currently support decimals and it didn't seem worth the time to handle this case.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/src/components/listings/search/LandingSearch.tsx
Comment (ColinBuyck) [inline sites/public/src/components/listings/search/LandingSearch.tsx:125]: Including updateValue in dependency array led to infinite calls

**Review by emilyjablonski — APPROVED** [review]
Screenshot 2023-08-02 at 4 43 16 PM

This work solves the issue's bug, and recognize that further work makes sense to be captured in a different ticket. The experience of focusing in/out of an input and the the currency/comma formatting disappearing feels a little unusual but nothing to block merge over.

One small note that also doesn't feel blocking to me but would be a nice-to-have is that once I enter a search from the home page and open the filter drawer, the number I entered is not currency formatted until I click into the input and then back out again (screenshot is first load).

**Comment by ColinBuyck** [comment]
Thanks for the review and I definitely agree that this is not ideal! Since resolving these pieces would require changes to UI-C, I don't think I will pursue them here but think that a general ticket to improve our formatting approach would be helpful. In verifying that this is the behavior across Bloom instances, I realized that filtering via rent is broken on Detroit for this very reason. Working on fix now 🏃 

https://github.com/metrotranscom/doorway/assets/53269332/12be8ddd-60d0-4f2f-8546-f1d7e41ff220

**Comment by ludtkemorgan** [comment]
After discussing this with Colin, I'd advocate for holding off on merging this. I think this solution works and is the best way to tackle it within the current setup but we might want to look at this from the long term perspective. Mostly because despite solving the problem it brings in a couple of new quarks to the flow and we might be refactoring this form in the nearish future depending on the uic audit findings

We should decide as a team if this new behavior is better or worse than having the existing bug in place. If it's better we can merge this PR and just make a note to clean it up when we do get around to implementing it the right way. However, if it is worse or no better I think we should wait until it is implemented the right way. Especially since I feel like the bug is very minor

**Comment by emilyjablonski** [comment]
I'm good either way! But a note that this sort of clean up is not stuff that will be captured in the audit.

**Pull request merged**
